File: app/src/main/java/vn/loitp/app/activity/customviews/layout/constraintlayout/constraintset/ConstraintSetActivity.java
package vn.loitp.app.activity.customviews.layout.constraintlayout.constraintset;

import android.os.Build;
import android.os.Bundle;
import android.transition.TransitionManager;
import android.view.View;

import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.constraintlayout.widget.ConstraintSet;

import com.core.base.BaseFontActivity;

import org.jetbrains.annotations.NotNull;

import vn.loitp.app.R;

public class ConstraintSetActivity extends BaseFontActivity {
    private static final String SHOW_BIG_IMAGE = "showBigImage";
    private boolean mShowBigImage = false;
    private ConstraintLayout mRootLayout;
    private ConstraintSet mConstraintSetNormal = new ConstraintSet();
    private ConstraintSet mConstraintSetBig = new ConstraintSet();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mRootLayout = findViewById(R.id.activity_constraintset_example);
        mConstraintSetNormal.clone(mRootLayout);
        mConstraintSetBig.load(this, R.layout.constraintset_example_big);
        if (savedInstanceState != null) {
            boolean previous = savedInstanceState.getBoolean(SHOW_BIG_IMAGE);
            if (previous != mShowBigImage) {
                mShowBigImage = previous;
                applyConfig();
            }
        }

        findViewById(R.id.imageView).setOnClickListener(this::toggleMode);
    }

    @Override
    protected boolean setFullScreen() {
        return false;
    }

    @Override
    protected String setTag() {
        return getClass().getSimpleName();
    }

    @Override
    protected int setLayoutResourceId() {
        return R.layout.activity_constraint_set;
    }

    @Override
    public void onSaveInstanceState(@NotNull Bundle outState) {
        super.onSaveInstanceState(outState);
        outState.putBoolean(SHOW_BIG_IMAGE, mShowBigImage);
    }

    public void toggleMode(View v) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
            TransitionManager.beginDelayedTransition(mRootLayout);
        }
        mShowBigImage = !mShowBigImage;
        applyConfig();
    }

    private void applyConfig() {
        if (mShowBigImage) {
            mConstraintSetBig.applyTo(mRootLayout);
        } else {
            mConstraintSetNormal.applyTo(mRootLayout);
        }
    }
}

File: app/src/main/java/vn/loitp/app/activity/customviews/layout/constraintlayout/constraintset/ConstraintSetActivity.kt
package vn.loitp.app.activity.customviews.layout.constraintlayout.constraintset

import android.annotation.SuppressLint
import android.os.Build
import android.os.Bundle
import android.transition.TransitionManager
import androidx.constraintlayout.widget.ConstraintSet
import androidx.core.view.isVisible
import com.loitpcore.annotation.IsFullScreen
import com.loitpcore.annotation.LogTag
import com.loitpcore.core.base.BaseFontActivity
import com.loitpcore.core.utilities.LUIUtil
import kotlinx.android.synthetic.main.activity_constraint_set.*
import vn.loitp.app.R

@LogTag("ConstraintSetActivity")
@IsFullScreen(false)
class ConstraintSetActivity : BaseFontActivity() {

    companion object {
        private const val SHOW_BIG_IMAGE = "showBigImage"
    }

    private var mShowBigImage = false
    private val mConstraintSetNormal = ConstraintSet()
    private val mConstraintSetBig = ConstraintSet()

    override fun setLayoutResourceId(): Int {
        return R.layout.activity_constraint_set
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        lActionBar.apply {
            LUIUtil.setSafeOnClickListenerElastic(
                view = this.ivIconLeft,
                runnable = {
                    onBackPressed()
                }
            )
            this.ivIconRight?.isVisible = false
            this.viewShadow?.isVisible = true
            this.tvTitle?.text = ConstraintSetActivity::class.java.simpleName
        }
        mConstraintSetNormal.clone(layoutConstrainRoot)
        mConstraintSetBig.load(this, R.layout.constraintset_example_big)
        if (savedInstanceState != null) {
            val previous = savedInstanceState.getBoolean(SHOW_BIG_IMAGE)
            if (previous != mShowBigImage) {
                mShowBigImage = previous
                applyConfig()
            }
        }
        tvContent.text = getString(R.string.large_text)
        imageView.setOnClickListener {
            toggleMode()
        }
    }

    public override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putBoolean(SHOW_BIG_IMAGE, mShowBigImage)
    }

    @SuppressLint("ObsoleteSdkInt")
    fun toggleMode() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
            TransitionManager.beginDelayedTransition(layoutConstrainRoot)
        }
        mShowBigImage = !mShowBigImage
        applyConfig()
    }

    private fun applyConfig() {
        if (mShowBigImage) {
            mConstraintSetBig.applyTo(layoutConstrainRoot)
        } else {
            mConstraintSetNormal.applyTo(layoutConstrainRoot)
        }
    }
}
